File: pset6/mario/more/mario.py

# Define recursion for pyramid
def block(layer):

    if layer == 0:
        return

    block(layer - 1)

    # Left-part of the pyramid
    print(" " * (spaceNum - layer), end="")
    print("#" * layer, end="")

    # Middle-part of the pyramid
    print(" " * 2, end="")

    # Right-part of the pyramid
    print("#" * layer, end="")
    print()


# Prompting the user to type in positive interger
while True:
    try:
        h = int(input("Height: "))

    # Re-prompting the user to type in valid interger
    except:
        continue

    if 1 <= h <= 8:
        break

# Build pyramid
spaceNum = h
block(h)

File: pset1/mario/more/mario.c
#include <cs50.h>
#include <stdio.h>


int main(void)
{
    //Prompt user for positive integer
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
 
 
    //Print out pyramid
    for (int i = 0; i < n; i++)
    {
        //Print out left-side space
        for (int j = 0; j < n - i - 1; j++)
        {
            printf(" ");
        }

        //Print out left-side # brick
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }
        
        //Print out spaces at middle
        for (int j = 0; j < 2; j++)
        {
            printf(" ");
        }
        
        //Print out right-side # brick
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }
        printf("\n");
    }
}